src/utils/normalize.ts

```diff
@@ -0,0 +1,94 @@
+import { add0x, isValidHexAddress } from '@metamask/utils';
+import type { Hex } from '@metamask/utils';
+import BN from 'bn.js';
+
+type EIP712Domain = {
+  verifyingContract: string;
+};
+
+type SignTypedMessageDataV3V4 = {
+  types: Record<string, unknown>;
+  domain: EIP712Domain;
+  primaryType: string;
+  message: unknown;
+};
+
+/**
+ * Normalizes the messageData for the eth_signTypedData
+ *
+ * @param messageData - The messageData to normalize.
+ * @returns The normalized messageData.
+ */
+export function normalizeTypedMessage(messageData: string) {
+  let data;
+  try {
+    data = parseTypedMessage(
+      messageData,
+    ) as unknown as SignTypedMessageDataV3V4;
```

Comment:
When using a type assertion it's good to know why it's necessary. After looking at `parseTypedMessage` I can see it's because of the `JSON.parse`. What are your thoughts on putting this type assertion in that function instead so it's more associated with the `JSON.parse` visually?

Reply:
That make sense @mcmire, fixed in https://github.com/MetaMask/eth-json-rpc-middleware/pull/309/commits/2ebe6855a76c9953f2054d108b606049a75a233e